IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate.Tests/ConditionalWaitTests.cs

```diff
@@ -1,32 +1,177 @@
 using System;
+using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
 using Xunit;
+using Xunit.Sdk;
 
 namespace IntelliTect.TestTools.Selenate.Tests
 {
     public class ConditionalWaitTests
     {
         [Fact]
-        public async Task CheckActionParamsForTypeChecking()
+        public async Task CheckActionParamsForInvalidTypeChecking()
         {
-            ConditionalWait wait = new ConditionalWait();
-            Exception ex = await Assert.ThrowsAsync<ArgumentException>(() => wait.WaitFor(TestVoidDelegate, TimeSpan.FromSeconds(1), typeof(string)));
-            Assert.Equal("Invalid type passed into exceptionsToIgnore paramter. Must be of type Exception.", ex.Message);
+            Exception ex = await Assert.ThrowsAsync<ArgumentException>(
+                () => Wait.Until(TestVoidDelegate, TimeSpan.FromSeconds(1), typeof(string)));
+            Assert.Equal("Invalid type passed into exceptionsToIgnore parameter. Must be of type Exception.", ex.Message);
         }
 
         [Fact]
-        public async Task CheckFuncParamsForTypeChecking()
+        public async Task CheckFuncParamsForInvalidTypeChecking()
         {
-            ConditionalWait wait = new ConditionalWait();
-            Exception ex = await Assert.ThrowsAsync<ArgumentException>(() => wait.WaitFor(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(string)));
-            Assert.Equal("Invalid type passed into exceptionsToIgnore paramter. Must be of type Exception.", ex.Message);
+            Exception ex = await Assert.ThrowsAsync<ArgumentException>(
+                () => Wait.Until(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(string)));
+            Assert.Equal("Invalid type passed into exceptionsToIgnore parameter. Must be of type Exception.", ex.Message);
         }
 
         [Fact]
-        public async Task CheckForUnexpectedException()
+        public async Task CheckActionParamsForMixedInvalidTypeChecking()
         {
-            ConditionalWait wait = new ConditionalWait();
-            await Assert.ThrowsAsync<NullReferenceException>(() => wait.WaitFor<ArgumentNullException>(() => TestNullRefException(), TimeSpan.FromSeconds(1)));
+            await Assert.ThrowsAsync<ArgumentException>(
+                () => Wait.Until(TestVoidDelegate, TimeSpan.FromSeconds(1), typeof(Exception), typeof(object)));
+        }
+
+        [Fact]
+        public async Task CheckFuncParamsForMixedInvalidTypeChecking()
+        {
+            await Assert.ThrowsAsync<ArgumentException>(
+                () => Wait.Until(TestReturnDelegate, TimeSpan.FromSeconds(1), typeof(object), typeof(Exception)));
+        }
+
+        [Fact]
+        public async Task CheckActionParamsForValidTypeChecking()
```

Comment:
It appears this test is all about asserting that `Wait.Until` swallows the passed in exception. The name of the test should reflect that. As a general rule, collapse all of your unit tests and just read the method names. If you can't tell what is being tested without expanding the unit test to look at its contents, consider renaming the method.